setup/src/Magento/Setup/Module/Di/Code/Reader/FileClassScanner.php

```diff
@@ -42,12 +43,8 @@ protected function extract()
         $triggerNamespace = false;
         $paramNestingLevel = $currentParamNestingLevel = 0;
         foreach ($tokens as $key => $token) {
-            if (!is_array($token)) {
-                if ($token == '{') {
-                    $paramNestingLevel++;
-                } else if ($token == '}') {
-                    $paramNestingLevel--;
-                }
+            if ($token == '{') {
+                return $classes;
             }
```

Comment:
Yeah, this code, which is slightly a hack to deal with the ::class issue, expects that the code people integrate follows the most basic of Magento coding standards.  I was also hoping that I wouldn't have to iterate over the entire file.  But if this is an important enough edge case (and I will presume it is unless told otherwise) then I will scan the whole file.